2017-2018/ROS_RoboBuggy/src/robobuggy/src/low_level/Transistor_LL_Broadcaster_Runner.cpp

```diff
@@ -0,0 +1,134 @@
+#include "transistor_serial_messages.h"
+#include "serial/serial.h"
+
+#include "low_level/Transistor_LL_Broadcaster.h"
+
+std::string rb_serial_buffer;
+
+int main(int argc, char **argv)
+{
+    ros::init(argc, argv, "Transistor_LL_Broadcaster");
+    ros::NodeHandle nh;
+
+    Transistor_LL_Broadcaster broadcaster;
+
+    // Messages
+    robobuggy::Brake brake_msg;
+    robobuggy::Steering steering_msg;
+    robobuggy::Diagnostics diagnostics_msg;
+    robobuggy::ENC encoder_msg;
+
+    // Read serial parameters from parameter server
+    std::string serial_port;
+    int serial_baud;
+    if (!nh.getParam("serial_port", serial_port)) {
+        ROS_INFO_STREAM("Serial port parameter not found, using default");
+        serial_port = "/dev/ttyACM1"; // Default value
+    }
+    if (!nh.getParam("serial_baud", serial_baud)) {
+        ROS_INFO_STREAM("Serial baud rate parameter not found, using default");
+        serial_baud = 57600; // Default value
+    }
+
+    // Initialize serial communication
+    serial::Serial rb_serial;
+    try {
+        rb_serial.setPort(serial_port);
+        rb_serial.setBaudrate(serial_baud);
+        serial::Timeout to = serial::Timeout::simpleTimeout(1000);
+        rb_serial.setTimeout(to);
+        rb_serial.open();
+    }
+    catch (serial::IOException& e) {
+        ROS_ERROR_STREAM("Unable to open port");
+        ROS_ERROR_STREAM(serial_port);
+        return -1;
+    }
+
+    if (rb_serial.isOpen()) {
+        ROS_INFO_STREAM("Serial port initialized");
+    }
+    else {
+        return -1;
+    }
+
+    //@TODO: Restructure this into the class file, instead of runner?
+
+    // Constantly read messages from low level
+    while(ros::ok()) {
+        if (rb_serial.available()) {
```

Comment:
This should be in the class, not the runner. The point of the runner is just to spin up the node.